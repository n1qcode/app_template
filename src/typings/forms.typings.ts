import * as yup from "yup";
import { ISchema } from "yup";

// @typescript-eslint/ban-ts-comment
// eslint-disable-next-line
export type FormDataType<T extends ISchema<any>> = yup.InferType<T>;
