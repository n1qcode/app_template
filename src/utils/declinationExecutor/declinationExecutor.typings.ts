export type PluralsType = [string, string, string];
