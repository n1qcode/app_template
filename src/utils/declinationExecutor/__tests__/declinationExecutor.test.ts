import declinationExecutor from "..";
import { PluralsType } from "../declinationExecutor.typings";

const plurals: PluralsType = ["tomato", "tomato", "tomatoes"];

const mock = [
  {
    description: "one",
    count: 1,
    result: plurals[0],
  },
  {
    description: "one",
    count: 21,
    result: plurals[0],
  },
  {
    description: "one",
    count: 31,
    result: plurals[0],
  },
  {
    description: "one",
    count: 41,
    result: plurals[0],
  },
  {
    description: "one",
    count: 101,
    result: plurals[0],
  },
  {
    description: "few",
    count: 2,
    result: plurals[1],
  },
  {
    description: "few",
    count: 3,
    result: plurals[1],
  },
  {
    description: "few",
    count: 4,
    result: plurals[1],
  },
  {
    description: "few",
    count: 54,
    result: plurals[1],
  },
  {
    description: "few",
    count: 104,
    result: plurals[1],
  },
  {
    description: "many",
    count: 0,
    result: plurals[2],
  },
  {
    description: "many",
    count: 5,
    result: plurals[2],
  },
  {
    description: "many",
    count: 6,
    result: plurals[2],
  },
  {
    description: "many",
    count: 50,
    result: plurals[2],
  },
  {
    description: "many",
    count: 110,
    result: plurals[2],
  },
  {
    description: "many",
    count: 500,
    result: plurals[2],
  },
];

describe("declinationExecutor", () => {
  test.each(mock)("%#: %o", (testCase) => {
    const { count, result } = testCase;
    expect(declinationExecutor(count, plurals)).toBe(result);
    expect(declinationExecutor(count, plurals)).toMatchSnapshot();
  });
});
