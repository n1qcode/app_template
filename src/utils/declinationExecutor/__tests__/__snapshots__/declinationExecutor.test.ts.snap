// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`declinationExecutor 0: { description: 'one', count: 1, result: 'tomato' } 1`] = `"tomato"`;

exports[`declinationExecutor 1: { description: 'one', count: 21, result: 'tomato' } 1`] = `"tomato"`;

exports[`declinationExecutor 2: { description: 'one', count: 31, result: 'tomato' } 1`] = `"tomato"`;

exports[`declinationExecutor 3: { description: 'one', count: 41, result: 'tomato' } 1`] = `"tomato"`;

exports[`declinationExecutor 4: { description: 'one', count: 101, result: 'tomato' } 1`] = `"tomato"`;

exports[`declinationExecutor 5: { description: 'few', count: 2, result: 'tomato' } 1`] = `"tomato"`;

exports[`declinationExecutor 6: { description: 'few', count: 3, result: 'tomato' } 1`] = `"tomato"`;

exports[`declinationExecutor 7: { description: 'few', count: 4, result: 'tomato' } 1`] = `"tomato"`;

exports[`declinationExecutor 8: { description: 'few', count: 54, result: 'tomato' } 1`] = `"tomato"`;

exports[`declinationExecutor 9: { description: 'few', count: 104, result: 'tomato' } 1`] = `"tomato"`;

exports[`declinationExecutor 10: { description: 'many', count: 0, result: 'tomatoes' } 1`] = `"tomatoes"`;

exports[`declinationExecutor 11: { description: 'many', count: 5, result: 'tomatoes' } 1`] = `"tomatoes"`;

exports[`declinationExecutor 12: { description: 'many', count: 6, result: 'tomatoes' } 1`] = `"tomatoes"`;

exports[`declinationExecutor 13: { description: 'many', count: 50, result: 'tomatoes' } 1`] = `"tomatoes"`;

exports[`declinationExecutor 14: { description: 'many', count: 110, result: 'tomatoes' } 1`] = `"tomatoes"`;

exports[`declinationExecutor 15: { description: 'many', count: 500, result: 'tomatoes' } 1`] = `"tomatoes"`;
